src/realm/parser/driver.cpp

```diff
@@ -1539,4 +1763,1380 @@ SubQuery<T> column(const Table& origin, ColKey origin_col_key, Query subquery)
+    out << ")";
+}
+void PrintingVisitor::visitRelational(RelationalNode& node)
+{
+    out << "RelationalNode(";
+    base::visitRelational(node);
+    out << ")";
+}
+void PrintingVisitor::visitStringOps(StringOpsNode& node)
+{
+    out << "StringOpsNode(";
+    base::visitStringOps(node);
+    out << ")";
+}
+void PrintingVisitor::visitTrueOrFalse(TrueOrFalseNode& node)
+{
+    out << "TrueOrFalseNode : value = " << node.true_or_false << " ";
+}
+void PrintingVisitor::visitPostOp(PostOpNode& node)
+{
+    out << "PostOpNode: opType = " << node.op_type << " opName = " << node.op_name << " ";
+}
+void PrintingVisitor::visitAggr(AggrNode& node)
+{
+    out << "AggrNode: type = " << node.type;
+}
+void PrintingVisitor::visitPath(PathNode& node)
+{
+    out << "PathNode: path_elems = {";
+    for (auto i : node.path_elems)
+        out << i << ", ";
+    out << "} ";
+}
+void PrintingVisitor::visitListAggr(ListAggrNode& node)
+{
+    out << "ListAggrNode(";
+    base::visitListAggr(node);
+    out << ")";
+}
+void PrintingVisitor::visitLinkAggr(LinkAggrNode& node)
+{
+    out << "LinkAggrNode(";
+    base::visitLinkAggr(node);
+    out << ")";
+}
+void PrintingVisitor::visitProp(PropNode& node)
+{
+    out << "PropNode(identifier = " << node.identifier << " comp_type: = " << static_cast<unsigned>(node.comp_type)
+        << " ";
+    base::visitProp(node);
+    out << ")";
+}
+void PrintingVisitor::visitSubquery(SubqueryNode& node)
+{
+    out << "SubQueryNode(";
+    base::visitSubquery(node);
+    out << ")";
+}
+void PrintingVisitor::visitDescriptor(DescriptorNode& node)
+{
+    out << "DescriptorNode, type = " << node.type;
+}
+void PrintingVisitor::visitDescriptorOrdering(DescriptorOrderingNode& node)
+{
+    out << "DescriptorOrderingNode(";
+    base::visitDescriptorOrdering(node);
+    out << ")";
+}
+
+Query QueryVisitor::visit(ParserNode& node)
+{
+    node.accept(*this);
+    return std::move(query);
+}
+
+void QueryVisitor::visitAnd(AndNode& node)
+{
+    auto& atom_preds = node.atom_preds;
+    if (atom_preds.size() == 1) {
+        atom_preds[0]->accept(*this);
+        return;
+    }
+    realm::Query q(drv->m_base_table);
+    for (auto& it : atom_preds) {
+        it->accept(*this);
+        q.and_query(query);
+    }
+    query = q;
+}
+
+void QueryVisitor::visitOr(OrNode& node)
+{
+    auto& atom_preds = node.atom_preds;
+    if (atom_preds.size() == 1) {
+        atom_preds[0]->accept(*this);
+        return;
+    }
+    realm::Query q(drv->m_base_table);
+    q.group();
+    for (auto& it : atom_preds) {
+        q.Or();
+        it->accept(*this);
+        q.and_query(query);
+    }
+    q.end_group();
+    query = q;
+}
+
+void QueryVisitor::visitNot(NotNode& node)
+{
+    node.atom_preds[0]->accept(*this);
+    realm::Query q = drv->m_base_table->where();
+    q.Not();
+    q.and_query(query);
+    query = q;
+}
+
+void QueryVisitor::visitEquality(EqualityNode& node)
+{
+    auto op = node.op;
```

Comment:
Ideally, common code like this would be extracted into a shared method that can be called by both places instead of copied here. That will both make it easier to read and enable better maintenance so that if something has to be fixed we don't have to remember to change two places.

Reply:
My bad on this one, the idea is to remove the old visit methods as the new NodeVisitor will supersede them